portal-api/src/v1/controllers/utilisation-report-service/utilisation-report-upload.controller.js

```diff
@@ -62,35 +64,74 @@ const sendEmailToBankPaymentOfficerTeam = async (reportPeriod, bankId, submitted
+ * azure storage details with folder & file name, full path & url.
+ */
+const saveFileToAzure = async (file, bankId) => {
+  try {
+    console.info(`Attempting to save utilisation report for bank: ${bankId}`);
+    const { originalname, buffer } = file;
+
+    const fileInfo = await uploadFile({
+      fileshare: FILESHARES.UTILISATION_REPORTS,
+      folder: bankId,
+      filename: formatFilenameForSharepoint(originalname),
+      buffer,
+      allowOverwrite: true,
+    });
+
+    console.info(`Successfully saved utilisation report for bank: ${bankId}`);
+    return { fileInfo, error: false };
+  } catch (error) {
+    console.error('Failed to save utilisation report: %O', error);
+    return { fileInfo: null, error: true };
+  }
+};
+
 const uploadReportAndSendNotification = async (req, res) => {
   try {
+    const { file } = req;
+
     const { reportPeriod, reportData, month, year, user } = req.body;
     const parsedReportData = JSON.parse(reportData);
     const parsedUser = JSON.parse(user);
 
-    // TODO: FN-967 save file to azure
-    // const file = req.file;
-
-    // if (!file) return res.status(400).send();
+    if (!file) return res.status(400).send();
 
     // If a report for this month/year/bank combo already exists we should not overwrite it
     const existingReports = await api.getUtilisationReports(parsedUser?.bank?.id, month, year);
     if (existingReports.length > 0) {
       return res.status(409).send('Report already exists');
     }
 
-    // const path = await saveFileToAzure(req.file, month, year, bank);
+    const { fileInfo, error } = await saveFileToAzure(file, parsedUser.bank.id);
+    if (error) {
+      const status = 500;
+      return res.status(status).send({ status, data: 'Failed to save utilisation report to Azure' });
+    }
+    const azureFileStorage = {
+      ...fileInfo,
+      mimetype: file.mimetype,
+    };
 
-    const saveDataResponse = await api.saveUtilisationReport(parsedReportData, month, year, parsedUser, 'a file path');
+    const saveDataResponse = await api.saveUtilisationReport(parsedReportData, month, year, parsedUser, azureFileStorage);
 
     if (saveDataResponse.status !== 201) {
       const status = saveDataResponse.status || 500;
       console.error('Failed to save utilisation report: %O', saveDataResponse);
       return res.status(status).send({ status, data: 'Failed to save utilisation report' });
     }
-    const submittedDateUtc = new Date().toISOString();
     await sendEmailToPdcInputtersEmail(parsedUser?.bank?.name, reportPeriod);
-    const { paymentOfficerEmail } = await sendEmailToBankPaymentOfficerTeam(reportPeriod, parsedUser?.bank?.id, submittedDateUtc, parsedUser);
+    const { paymentOfficerEmail } = await sendEmailToBankPaymentOfficerTeam(
+      reportPeriod,
+      parsedUser?.bank?.id,
+      new Date(saveDataResponse.data.dateUploaded),
```

Comment:
I think the `sendEmailToBankPaymentOfficerTeam` is expecting a string here, though it might make more sense to pass the date around instead (as you've done) until we need it formatted. If we keep this change we should also update the type of the function and update where the value is used to use a Date instead of a string.